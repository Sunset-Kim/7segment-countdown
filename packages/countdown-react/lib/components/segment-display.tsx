import { useId } from "react";
import { SevenSegment, SevenSegmentProps } from "./seven-segment";

interface SegmentDisplayProps extends Omit<SevenSegmentProps, "digit"> {
  value: number;
  place?: number;
}

export function SegmentDisplay({
  value,
  place = 2,
  ...props
}: SegmentDisplayProps) {
  const id = useId();
  return value
    .toString()
    .padStart(place, "0")
    .split("")
    .map((digit, index) => (
      <SevenSegment key={id + index} digit={parseInt(digit)} {...props} />
    ));
}
