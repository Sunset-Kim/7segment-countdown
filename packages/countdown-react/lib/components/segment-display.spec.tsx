import { describe, it, expect, beforeEach } from "vitest";
import { render, screen, cleanup } from "@testing-library/react";
import { SegmentDisplay } from "./segment-display";

describe("SegmentDisplay", () => {
  beforeEach(() => {
    cleanup();
  });

  it("기본 props로 렌더링", () => {
    const { container } = render(<SegmentDisplay value={42} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("다양한 값과 place로 렌더링", () => {
    const { container } = render(<SegmentDisplay value={123} place={4} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("value가 올바르게 표시되는지 확인", () => {
    render(<SegmentDisplay value={42} />);

    expect(screen.getAllByRole("img")).toHaveLength(2);
  });

  it("place에 따라 0이 패딩되는지 확인", () => {
    render(<SegmentDisplay value={5} place={3} />);
    expect(screen.getAllByRole("img")).toHaveLength(3);
  });

  it("0 값 처리", () => {
    render(<SegmentDisplay value={0} />);
    expect(screen.getAllByRole("img")).toHaveLength(2);
  });

  it("큰 숫자 처리", () => {
    render(<SegmentDisplay value={9999999} place={5} />);
    expect(screen.getAllByRole("img")).toHaveLength(7);
  });
});
