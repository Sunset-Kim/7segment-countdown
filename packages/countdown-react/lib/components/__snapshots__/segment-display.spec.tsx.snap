// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SegmentDisplay > 기본 props로 렌더링 1`] = `
<svg
  height="200"
  role="img"
  viewBox="0 0 100 200"
  width="100"
>
  <g>
    <polygon
      class="segment "
      data-testid="segment-A"
      id="segment-A"
      points="10,10 20,2.5 80,2.5
   90,10 80,17.5 20,17.5"
      style="fill: aliceblue; transition: fill 300ms;"
    />
    <polygon
      class="segment on"
      data-testid="segment-B"
      id="segment-B"
      points="90,10 97.5,20 97.5,80
   90,90 82.5,80 82.5,20"
      style="fill: dodgerblue; transition: fill 300ms;"
    />
    <polygon
      class="segment on"
      data-testid="segment-C"
      id="segment-C"
      points="90,90 97.5,100 97.5,160
   90,170 82.5,160 82.5,100"
      style="fill: dodgerblue; transition: fill 300ms;"
    />
    <polygon
      class="segment "
      data-testid="segment-D"
      id="segment-D"
      points="10,170 20,162.5 80,162.5
   90,170 80,177.5 20,177.5"
      style="fill: aliceblue; transition: fill 300ms;"
    />
    <polygon
      class="segment "
      data-testid="segment-E"
      id="segment-E"
      points="10,90 17.5,100 17.5,160
   10,170 2.5,160 2.5,100"
      style="fill: aliceblue; transition: fill 300ms;"
    />
    <polygon
      class="segment on"
      data-testid="segment-F"
      id="segment-F"
      points="10,10 17.5,20 17.5,80
   10,90 2.5,80 2.5,20"
      style="fill: dodgerblue; transition: fill 300ms;"
    />
    <polygon
      class="segment on"
      data-testid="segment-G"
      id="segment-G"
      points="10,90 20,82.5 80,82.5
   90,90 80,97.5 20,97.5"
      style="fill: dodgerblue; transition: fill 300ms;"
    />
  </g>
</svg>
`;

exports[`SegmentDisplay > 다양한 값과 place로 렌더링 1`] = `
<svg
  height="200"
  role="img"
  viewBox="0 0 100 200"
  width="100"
>
  <g>
    <polygon
      class="segment on"
      data-testid="segment-A"
      id="segment-A"
      points="10,10 20,2.5 80,2.5
   90,10 80,17.5 20,17.5"
      style="fill: dodgerblue; transition: fill 300ms;"
    />
    <polygon
      class="segment on"
      data-testid="segment-B"
      id="segment-B"
      points="90,10 97.5,20 97.5,80
   90,90 82.5,80 82.5,20"
      style="fill: dodgerblue; transition: fill 300ms;"
    />
    <polygon
      class="segment on"
      data-testid="segment-C"
      id="segment-C"
      points="90,90 97.5,100 97.5,160
   90,170 82.5,160 82.5,100"
      style="fill: dodgerblue; transition: fill 300ms;"
    />
    <polygon
      class="segment on"
      data-testid="segment-D"
      id="segment-D"
      points="10,170 20,162.5 80,162.5
   90,170 80,177.5 20,177.5"
      style="fill: dodgerblue; transition: fill 300ms;"
    />
    <polygon
      class="segment on"
      data-testid="segment-E"
      id="segment-E"
      points="10,90 17.5,100 17.5,160
   10,170 2.5,160 2.5,100"
      style="fill: dodgerblue; transition: fill 300ms;"
    />
    <polygon
      class="segment on"
      data-testid="segment-F"
      id="segment-F"
      points="10,10 17.5,20 17.5,80
   10,90 2.5,80 2.5,20"
      style="fill: dodgerblue; transition: fill 300ms;"
    />
    <polygon
      class="segment "
      data-testid="segment-G"
      id="segment-G"
      points="10,90 20,82.5 80,82.5
   90,90 80,97.5 20,97.5"
      style="fill: aliceblue; transition: fill 300ms;"
    />
  </g>
</svg>
`;
